lib/manageiq/api/common/user.rb

```diff
@@ -0,0 +1,87 @@
+module ManageIQ
+  module API
+    module Common
+      class User
+        IDENTITY_KEY = 'x-rh-identity'.freeze
+
+        def self.current
+          Thread.current[:attr_current_user]
+        end
+
+        def self.current=(user_header)
+          Thread.current[:attr_current_user] = new(user_header)
+        end
+
+        def self.with_logged_in_user(user_header)
+          self.current = user_header
+          yield self.current
+        ensure
+          Thread.current[:attr_current_user] = nil
+        end
+
+        def initialize(user_header)
+          @user = user_header
+        end
+
+        def username
+          validate_and_return(__method__)
+        end
+
+        def email
+          validate_and_return(__method__)
+        end
+
+        def first_name
+          validate_and_return(__method__)
+        end
+
+        def last_name
+          validate_and_return(__method__)
+        end
+
+        def is_active?
+          validate_and_return(:is_active)
+        end
+
+        def is_org_admin?
+          validate_and_return(:is_org_admin)
+        end
+
+        def is_internal?
+          validate_and_return(:is_internal)
+        end
+
+        def locale
+          validate_and_return(__method__)
+        end
+
+        def tenant
+          validate_and_return(:account_number)
+        end
+
+        private
+
+        def decode
+          if @user.is_a?(ActionDispatch::Http::Headers)
+            val = @user.instance_variable_get(:@req).stringify_keys
```

Comment:
The instance_variable_get feels weird to me.

Reply:
I saw something in @mkanoor 's PR, and I think this code will actually break outside of tests as is.  When you use `ActionDispatch::Http::Headers.new`, it's expecting a request object as the parameter, not a Hash.  As such, pulling out `@req` in production is invalid (might work by happenstance, but not the right approach).

Instead of using `.new` in the tests, you can use `.from_hash`, and then you can just call `ActionDispatch::Http::Headers#to_h` after that to get the headers back out.  The .to_h only works when you initialize it in this more accurate way.